aries_cloudagent/messaging/jsonld/routes.py

```diff
@@ -93,25 +99,57 @@ async def verify(request: web.BaseRequest):
     response = {"valid": False}
     try:
         context: AdminRequestContext = request["context"]
+        profile = context.profile
         body = await request.json()
         verkey = body.get("verkey")
         doc = body.get("doc")
-
         async with context.session() as session:
-            wallet = session.inject(BaseWallet, required=False)
-            if not wallet:
+            try:
+                session.inject(BaseWallet, required=True)
+            except Exception:
                 raise web.HTTPForbidden(reason="No wallet available")
-            valid = await verify_credential(doc, verkey, wallet)
-
+            if verkey is None:
+                resolver = session.inject(DIDResolver)
+                ver_meth_expanded = await resolver.dereference(
+                    profile, doc["proof"]["verificationMethod"]
+                )
+                if ver_meth_expanded is None:
+                    raise ResolverError(
+                        f"Verification method "
+                        f"{doc['proof']['verificationMethod']} not found."
+                    )
+                verkey = ver_meth_expanded.material
+            valid = await verify_credential(session, doc, verkey)
         response["valid"] = valid
-    except (BadJWSHeaderError, DroppedAttributeError) as e:
-        response["error"] = str(e)
+    except (
+        BadJWSHeaderError,
+        DroppedAttributeError,
+        DIDError,
```

Comment:
We shouldn't have to directly handle PyDID errors; we should wrap this error from the resolver interfaces with an error subclassing `ResolverError`.

Reply:
I will fix.